### PR TITLE
Fix golint ppc64le issues and eventually Travis-CI

### PR DESCRIPTION
**Pull request by nitkon**

### CODE REVIEW

**Comment by katabuilder** [comment]
PSS Measurement:
Qemu: 162010 KB
Proxy: 6782 KB
Shim: 8740 KB

Memory inside container:
Total Memory: 2045972 KB
Free Memory:  1996880 KB

**Comment by nitkon** [comment]
Travis-CI on ppc64le
```
15.19s$ make
     CLEAN   clean
     GENERATE cli/config-generated.go
     CONFIG  data/kata-collect-data.sh
kata-runtime - version 1.0.0 (commit d6b956a1aac5c9a08a86db49fd3f0e66c02363f8)
• architecture:
	Host: ppc64le
	golang: ppc64le
	Build: ppc64le
• golang:
	go version go1.8 linux/ppc64le
• Summary:
	binary install path (DESTTARGET)      : /usr/local/bin/kata-runtime
	config install path (DESTCONFIG)      : /usr/share/defaults/kata-containers/configuration.toml
	alternate config path (DESTSYSCONFIG) : /etc/kata-containers/configuration.toml
	hypervisor path (QEMUPATH)            : /usr/bin/qemu-system-ppc64le
	assets path (PKGDATADIR)              : /usr/share/kata-containers
	proxy+shim path (PKGLIBEXECDIR)       : /usr/libexec/kata-containers
     BUILD   /home/travis/gopath/src/github.com/kata-containers/runtime/kata-runtime
     CONFIG  cli/config/configuration.toml
install.3
0.57s$ sudo -E PATH=$PATH make install
kata-runtime - version 1.0.0 (commit d6b956a1aac5c9a08a86db49fd3f0e66c02363f8)
• architecture:
	Host: ppc64le
	golang: ppc64le
	Build: ppc64le
• golang:
	go version go1.8 linux/ppc64le
• Summary:
	binary install path (DESTTARGET)      : /usr/local/bin/kata-runtime
	config install path (DESTCONFIG)      : /usr/share/defaults/kata-containers/configuration.toml
	alternate config path (DESTSYSCONFIG) : /etc/kata-containers/configuration.toml
	hypervisor path (QEMUPATH)            : /usr/bin/qemu-system-ppc64le
	assets path (PKGDATADIR)              : /usr/share/kata-containers
	proxy+shim path (PKGLIBEXECDIR)       : /usr/libexec/kata-containers
     INSTALL install-scripts
     INSTALL install
     INSTALL install
before_script
85.40s$ .ci/static-checks.sh
make -C cmd/checkcommits
make[1]: Entering directory '/home/travis/gopath/src/github.com/kata-containers/tests/cmd/checkcommits'
go test .
ok  	github.com/kata-containers/tests/cmd/checkcommits	0.010s
go install -ldflags "-X main.appCommit="3d74611ed16393c9e223b22f8e38d08582da3211" -X main.appVersion=0.0.1" .
make[1]: Leaving directory '/home/travis/gopath/src/github.com/kata-containers/tests/cmd/checkcommits'
Running checkcommits version 0.0.1 (commit 3d74611ed16393c9e223b22f8e38d08582da3211)
Detected TravisCI Environment
Defaulting commit to HEAD
Found 0 commits between commit HEAD and branch golint
All commit checks passed.
INFO: Checking for SPDX license headers
fatal: ambiguous argument 'origin/master': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
INFO: No files found
comm: file 1 is not in sorted order
INFO: Installing gometalinter
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100     9    0     9    0     0     36      0 --:--:-- --:--:-- --:--:--    36
/home/travis/gopath/bin/yq: line 1: Not: command not found
INFO: Forcing gometalinter version 
Installing:
  deadcode
  dupl
  errcheck
  gas
  goconst
  gocyclo
  goimports
  golint
  gosimple
  gotype
  gotypex
  ineffassign
  interfacer
  lll
  maligned
  megacheck
  misspell
  nakedret
  safesql
  staticcheck
  structcheck
  unconvert
  unparam
  unused
  varcheck
INFO: gometalinter args: '--exclude=vendor/.*" --tests --exclude="protocols/grpc/.*\.pb\.go" --disable-all --enable=gofmt --enable=misspell --enable=vet --enable=ineffassign --enable=gocyclo --cyclo-over=15 --enable=golint --deadline=600s --enable=structcheck --enable=unused --enable=staticcheck --enable=maligned --enable=varcheck --enable=unconvert'
     GOBUILD  build
     GOBUILD  virtc
     GOBUILD  hook
     GOBUILD  cc-shim
     GOBUILD  kata-shim
INFO: Running gometalinter checks on the following packages:
github.com/kata-containers/runtime/cli
github.com/kata-containers/runtime/virtcontainers
github.com/kata-containers/runtime/virtcontainers/device/api
github.com/kata-containers/runtime/virtcontainers/device/config
github.com/kata-containers/runtime/virtcontainers/device/drivers
github.com/kata-containers/runtime/virtcontainers/device/manager
github.com/kata-containers/runtime/virtcontainers/hack/virtc
github.com/kata-containers/runtime/virtcontainers/hook/mock
github.com/kata-containers/runtime/virtcontainers/pkg/annotations
github.com/kata-containers/runtime/virtcontainers/pkg/annotations/dockershim
github.com/kata-containers/runtime/virtcontainers/pkg/cni
github.com/kata-containers/runtime/virtcontainers/pkg/hyperstart
github.com/kata-containers/runtime/virtcontainers/pkg/hyperstart/mock
github.com/kata-containers/runtime/virtcontainers/pkg/mock
github.com/kata-containers/runtime/virtcontainers/pkg/nsenter
github.com/kata-containers/runtime/virtcontainers/pkg/oci
github.com/kata-containers/runtime/virtcontainers/pkg/uuid
github.com/kata-containers/runtime/virtcontainers/pkg/vcmock
github.com/kata-containers/runtime/virtcontainers/shim/mock/cc-shim
github.com/kata-containers/runtime/virtcontainers/shim/mock/kata-shim
github.com/kata-containers/runtime/virtcontainers/utils
INFO: Package paths:
/home/travis/gopath/src/github.com/kata-containers/runtime/cli
/home/travis/gopath/src/github.com/kata-containers/runtime/virtcontainers
/home/travis/gopath/src/github.com/kata-containers/runtime/virtcontainers/device/api
/home/travis/gopath/src/github.com/kata-containers/runtime/virtcontainers/device/config
/home/travis/gopath/src/github.com/kata-containers/runtime/virtcontainers/device/drivers
/home/travis/gopath/src/github.com/kata-containers/runtime/virtcontainers/device/manager
/home/travis/gopath/src/github.com/kata-containers/runtime/virtcontainers/hack/virtc
/home/travis/gopath/src/github.com/kata-containers/runtime/virtcontainers/hook/mock
/home/travis/gopath/src/github.com/kata-containers/runtime/virtcontainers/pkg/annotations
/home/travis/gopath/src/github.com/kata-containers/runtime/virtcontainers/pkg/annotations/dockershim
/home/travis/gopath/src/github.com/kata-containers/runtime/virtcontainers/pkg/cni
/home/travis/gopath/src/github.com/kata-containers/runtime/virtcontainers/pkg/hyperstart
/home/travis/gopath/src/github.com/kata-containers/runtime/virtcontainers/pkg/hyperstart/mock
/home/travis/gopath/src/github.com/kata-containers/runtime/virtcontainers/pkg/mock
/home/travis/gopath/src/github.com/kata-containers/runtime/virtcontainers/pkg/nsenter
/home/travis/gopath/src/github.com/kata-containers/runtime/virtcontainers/pkg/oci
/home/travis/gopath/src/github.com/kata-containers/runtime/virtcontainers/pkg/uuid
/home/travis/gopath/src/github.com/kata-containers/runtime/virtcontainers/pkg/vcmock
/home/travis/gopath/src/github.com/kata-containers/runtime/virtcontainers/shim/mock/cc-shim
/home/travis/gopath/src/github.com/kata-containers/runtime/virtcontainers/shim/mock/kata-shim
/home/travis/gopath/src/github.com/kata-containers/runtime/virtcontainers/utils
INFO: Checking documentation
INFO: processing file './README.md'
INFO: processing file './virtcontainers/README.md'
INFO: processing file './virtcontainers/hack/virtc/README.md'
INFO: processing file './virtcontainers/documentation/api/1.0/api.md'
INFO: processing file './virtcontainers/documentation/Developers.md'
INFO: processing file './CONTRIBUTING.md'
INFO: processing file './CODE_OF_CONDUCT.md'
INFO: processing file './.github/ISSUE_TEMPLATE.md'
3.31s$ make
     CLEAN   clean
     GENERATE cli/config-generated.go
     CONFIG  data/kata-collect-data.sh
kata-runtime - version 1.0.0 (commit d6b956a1aac5c9a08a86db49fd3f0e66c02363f8)
• architecture:
	Host: ppc64le
	golang: ppc64le
	Build: ppc64le
• golang:
	go version go1.8 linux/ppc64le
• Summary:
	binary install path (DESTTARGET)      : /usr/local/bin/kata-runtime
	config install path (DESTCONFIG)      : /usr/share/defaults/kata-containers/configuration.toml
	alternate config path (DESTSYSCONFIG) : /etc/kata-containers/configuration.toml
	hypervisor path (QEMUPATH)            : /usr/bin/qemu-system-ppc64le
	assets path (PKGDATADIR)              : /usr/share/kata-containers
	proxy+shim path (PKGLIBEXECDIR)       : /usr/libexec/kata-containers
     BUILD   /home/travis/gopath/src/github.com/kata-containers/runtime/kata-runtime
     CONFIG  cli/config/configuration.toml
The command "make" exited with 0.
Done. Your build exited with 0.
```

File: cli/kata-check_ppc64le.go [hunk 1]
```diff
@@ -44,6 +44,12 @@ func archHostCanCreateVMContainer() error {
 // hostIsVMContainerCapable checks to see if the host is theoretically capable
 // of creating a VM container.
 func hostIsVMContainerCapable(details vmContainerCapableDetails) error {
+
+	_, err := getCPUInfo(details.cpuInfoFile)
```
Comment (jodh-intel) [inline cli/kata-check_ppc64le.go]: I'm not clear why this call is needed here?

Comment (nitkon) [reply cli/kata-check_ppc64le.go]: @jodh-intel : Else, hostIsVMContainerCapable returns "nil" on ppc64le, resulting in testcase " TestCheckCLIFunctionFail" failure as it expects an error. This is silimar to "func genericHostIsVMContainerCapable" from cli/kata-check.go +367

Comment (jodh-intel) [reply cli/kata-check_ppc64le.go]: Ah right.

lgtm

**Comment by jodh-intel** [comment]
lgtm

File: cli/kata-check_ppc64le.go (hunk 1 above)
Comment (jodh-intel) [reply cli/kata-check_ppc64le.go]: Hmm, so do we need an update for ARM64 I wonder as it uses the same code?

/cc @Pennyzct.

Comment (jodh-intel) [reply cli/kata-check_ppc64le.go]: Well, not the same code but the same logic in `hostIsVMContainerCapable()`.

Comment (Pennyzct) [reply cli/kata-check_ppc64le.go]: @jodh-intel thanks for raising up. we will give an update asap😊.

**Pull request merged**